stdlib/LibGit2/src/LibGit2.jl

```diff
@@ -275,8 +275,9 @@ function fetch(repo::GitRepo; remote::AbstractString="origin",
         GitRemoteAnon(repo, remoteurl)
     end
     result = try
-        fo = FetchOptions(callbacks=RemoteCallbacks(credentials=credentials_cb(), payload=p))
-        fetch(rmt, refspecs, msg="from $(url(rmt))", options = fo)
+        callbacks = RemoteCallbacks(credentials=(credentials_cb(), pointer_from_objref(p)))
```

Comment:
`pointer_from_objref` is fishy here since it allows `p` to be gc'd, 
that's why original code uses `Ref`.

Reply:
`GC.@preserve` it?

Reply:
Too brittle and error-prone. This code really just needs to avoid using `Ptr{Void}` and stick with `Ref`.